morf-testsupport/src/main/java/org/alfasoftware/morf/jdbc/AbstractSqlDialectTestWithNoSupportForWindowFunctions.java

```diff
@@ -0,0 +1,41 @@
+package org.alfasoftware.morf.jdbc;
+
+import org.hamcrest.Matchers;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+/**
+ * @author Copyright (c) Alfa Financial Software 2022
+ */
+public abstract class AbstractSqlDialectTestWithNoSupportForWindowFunctions extends AbstractSqlDialectTest{
+
+    protected final void verifyExceptionForNoOrderByClause(Exception e) {
+        verifyWindowFunctionNotSupported(e);
+    }
+
+    @Override
+    protected void verifyNoExceptionForWithoutPartitionClause(String result) {
+        fail("Window function not supported. UnsupportedOperationException should be thrown");
+    }
+
+    @Override
+    protected void verifyException(Exception e) {
+        verifyWindowFunctionNotSupported(e);
+    }
+
+    @Override
+    protected void verifyNoExceptionForWithPartitionClause(String result) {
+        fail("Window function not supported. UnsupportedOperationException should be thrown");
+    }
+
+    @Override
+    protected void verifyNoExceptionForWithoutOrderByClause() {
+        fail("Window function not supported. UnsupportedOperationException should be thrown");
+    }
+
+    private void verifyWindowFunctionNotSupported(Exception e){
+        assertThat(e, Matchers.instanceOf(UnsupportedOperationException.class));
+        assertThat(e.getMessage(), Matchers.containsString("does not support window functions"));
+    }
+}
```

Comment:
Given the fact that you can only extend from one abstract class, this is not the best choice.
For example, thee are DbLinks, which are also not supported on H2.

Reply:
Just saw how it is done for dblinks. I was not aware of the assume function used, that could be used to cleverly ignore the test since it is unsupported. At the moment (See other comment), this may not be required as I think I can remove unsupported exception in all the DBs that are currently showing as unsupported.